cordapp-cpk/src/main/kotlin/net/corda/plugins/cpb/CpbTask.kt

```diff
@@ -0,0 +1,76 @@
+package net.corda.plugins.cpb
+
+import net.corda.plugins.cpk.CORDA_CPK_TYPE
+import net.corda.plugins.cpk.CPK_FILE_EXTENSION
+import net.corda.plugins.cpk.SignJar.Companion.sign
+import net.corda.plugins.cpk.Signing
+import org.gradle.api.file.DuplicatesStrategy
+import org.gradle.api.tasks.AbstractCopyTask
+import org.gradle.api.tasks.bundling.Jar
+import org.gradle.api.tasks.bundling.ZipEntryCompression
+import java.nio.file.Files
+import java.util.jar.JarInputStream
+import java.util.zip.ZipInputStream
+
+open class CpbTask : Jar() {
+
+    companion object {
+        private const val CPB_ARTIFACT_CLASSIFIER = "package"
+        const val CPB_FILE_EXTENSION = "cpb"
+        private const val CPK_FILE_SUFFIX = ".$CPK_FILE_EXTENSION"
+        private const val JAR_FILE_SUFFIX = ".jar"
+        private val EXCLUDED_CPK_TYPES = setOf("corda-api")
+    }
+
+    val signing : Signing = project.objects.newInstance(Signing::class.java)
```

Comment:
I am assuming that the CPB will want to use the same signing information as both the "main" jar and the CPK, because the CorDapp developers are not going to want to configure it twice. You should be able to do what the CPK task does, i,e, add a `doLast` handler when the plugin configures the new `cpbTask`.

And because the `cordapp.signing` object is now part of the `cpbTask`'s input properties, you also need to declare it as such:
```kotlin
val cpbTaskProvider = project.tasks.register(CPB_TASK_NAME, CpbTask::class.java) { cpbTask ->
    cpbTask.inputs.nested("signing", cordapp.signing)
    cpbTask.from(allCPKs)
    ...
```
This guarantees that Gradle knows to rebuild the CPB if the signing block changes.

Reply:
mmm... I think it is better to provide the flexibility to sign the cpb with a different key but I could use the `Signing` object of the cordapp extension as the default. The idea is that you should be able to use this task to create a cpb with whatever signature you want and whatever cpk (even one that comes from a different project or  an external Maven repo)

Reply:
People are creating their _own_ CPB here. Other people's CPK will be downloaded from remote repositories and will already have been signed. Please just use the existing mechanism for the sake of consistency.